docs/deploy/nginx-uwsgi.rst

```diff
@@ -0,0 +1,246 @@
+.. _deploy:
+
+
+Deploying Falcon (Linux + NGINX + uWSGI)
+========================================
+
+
+NGINX is a powerful web server and reverse proxy and uWSGI is a fast and
+highly-configurable WSGI application server. Together, NIGNX and uWSGI create a
+one-two punch of speed and functionality which will suffice for most
+applications. In addition, this stack provides the building blocks for a
+horizontally-scalable and highly-available (HA) production environment. The
+configuration below does not go that far, but it is certainly possible.
+
+This guide provides instructions for deploying to a Linux environment only.
+However, with a bit of effort you should be able to adapt this configuration to
+other operating systems, such as OpenBSD or Windows.
+
+
+Running your Application as a Different User
+''''''''''''''''''''''''''''''''''''''''''''
+
+It is best to execute the application as a different OS user than the one who
+owns the source code for your application. The application user should *NOT*
+have write access to your source. This mitigates the chance that someone could
+write a malicious Python file to your source directory through an upload
+endpoint you might define; when your application restarts, the malicious file is
+loaded and proceeds to cause any number of BadThings\ :sup:(tm) to happen.
+
+.. code:: sh
+
+  $ useradd myproject --create-home
+  $ useradd myproject-runner --no-create-home
+
+It is helpful to switch to the project user (myproject) and use the home
+directory as the application environment.
+
+If you are working on a remote server, switch to the myproject user and pull
+down the source code for your application.
+
+.. code:: sh
+
+  $ git clone git@github.com/myorg/myproject.git /home/myproject/src
+
+
+.. note::
+
+  You could use a tarball, zip file, scp or any other means to get your source
+  onto a server.
+
+Then create a virtual environment which can be used to install your
+dependencies.
+
+.. code:: sh
+
+  # For Python 3
+  $ python3 -m venv /home/myproject/venv
+
+  # For Python 2
+  $ virtualenv /home/myproject/venv
+
+Then install your dependencies.
+
+.. code:: sh
+
+  $ /home/myproject/venv/bin/pip install -r /home/myproject/src/requirements.txt
+  $ /home/myproject/venv/bin/pip install -e /home/myproject/src
+  $ /home/myproject/venv/bin/pip install uwsgi
+
+
+.. note::
+
+  The exact commands for creating a virtual environment might differ based on
+  the Python version you are using and the system you are on. At the end of the day
+  we need a virtualenv in /home/myproject/venv with your project dependencies
```

Comment:
we -> you? Or perhaps leaving pronouns out and stating "the applications needs a virtualenv"

There are a lot of "you"s below, so staying consistent with the second person is best.